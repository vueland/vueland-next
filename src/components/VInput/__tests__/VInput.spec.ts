import { mount, VueWrapper } from '@vue/test-utils'
import { VInput } from '../index'
import 'regenerator-runtime/runtime'

describe('VInput', () => {
  let mountFunction: (options?: any) => VueWrapper<any>

  beforeEach(() => {
    mountFunction = (options = {}) => mount(VInput, {
      ...options,
      global: {
        provide: {
          $options: null
        }
      }
    })
  })

  test('should mount component and match snapshot', () => {
    const wrapper = mountFunction()
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('should test "focused" prop and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        focused: true,
      }
    })

    expect(wrapper.attributes().class).toContain('v-input--focused')
    expect(wrapper.attributes().class).toContain('v-input--dirty')
    expect(wrapper.html()).toMatchSnapshot()
  })
})
