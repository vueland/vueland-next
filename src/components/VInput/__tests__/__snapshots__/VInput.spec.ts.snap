// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VInput should mount component and match snapshot 1`] = `
<div class="v-input v-input--primary">
  <div class="v-input__label">
    <label class="v-label v-label--on-input">
    </label>
  </div>
  <div class="v-input__text-field">
  </div>
  <div class="v-input__hints">
    <transition-stub>
    </transition-stub>
  </div>
</div>
`;
